os/fs/driver/mtd/smart.c

```diff
@@ -513,14 +513,41 @@ static const struct block_operations g_bops = {
  * Public Functions
  ****************************************************************************/
 
+#ifdef CONFIG_DEBUG_FS
+/****************************************************************************
+ * Name: smart_retrieve_sector
+ *
+ * Description: Write specific sector data to buffer and send back.
+ *
+ ****************************************************************************/
+
+static int smart_retrieve_sector(FAR struct smart_struct_s *dev, FAR char* buffer)
+{
+	int ret = OK;
+	uint16_t sector = 0;
+
+	sector = UINT8TOUINT16(buffer[0]);
+	sector = (sector << 8);
+	sector += UINT8TOUINT16(buffer[1]);
+	if (sector >= (dev->geo.neraseblocks * dev->sectorsPerBlk)) {
+		return -EINVAL;
+	}
+
+	ret = MTD_BREAD(dev->mtd, sector * dev->mtdBlksPerSector, dev->mtdBlksPerSector, (FAR uint8_t *)buffer);
+	if (ret != dev->mtdBlksPerSector) {
+		return -EIO;
+	}
+
+	return CONFIG_MTD_SMART_SECTOR_SIZE;
```

Comment:
This value is not related to above MTD_BREAD, it had better return dev->mtdBlkPerSector * dev->sectorsize
anyway it is aligned to Todo in line 362 of crashdump.c
And please add more info to description regarding return value.

Reply:
Updated